error on build

@claude resolve these errors flagged by my deploy.yml workflow:
```
Run echo "Running type checking..."
Running type checking...
> openwebui-config-tool@1.0.0 typecheck
> tsc --noEmit
Error: src/components/ui/form/feedback/validation-summary.tsx(56,11): error TS2322: Type '{ value: number; className: string; indicatorClassName: string; }' is not assignable to type 'IntrinsicAttributes & Omit<ProgressProps & RefAttributes<HTMLDivElement>, "ref"> & RefAttributes<HTMLDivElement>'.
  Property 'indicatorClassName' does not exist on type 'IntrinsicAttributes & Omit<ProgressProps & RefAttributes<HTMLDivElement>, "ref"> & RefAttributes<HTMLDivElement>'.
Error: src/form/custom-fields/OverrideableField.tsx(2,10): error TS2614: Module '"@rjsf/core"' has no exported member 'FieldProps'. Did you mean to use 'import FieldProps from "@rjsf/core"' instead?
Error: src/form/custom-fields/OverrideableField.tsx(28,7): error TS2322: Type '{ children: CElement<any, Component<any, any, any>>; defaultValue: any; onOverride: (overridden: boolean) => void; }' is not assignable to type 'IntrinsicAttributes & OverrideableFieldProps'.
  Property 'onOverride' does not exist on type 'IntrinsicAttributes & OverrideableFieldProps'. Did you mean 'overridden'?
Error: src/form/custom-fields/PlanRestrictedField.tsx(2,10): error TS2614: Module '"@rjsf/core"' has no exported member 'FieldProps'. Did you mean to use 'import FieldProps from "@rjsf/core"' instead?
Error: src/form/custom-fields/PlanRestrictedField.tsx(16,28): error TS2322: Type '{ children: CElement<any, Component<any, any, any>>; planRequired: string; }' is not assignable to type 'IntrinsicAttributes & PlanRestrictedFeatureProps'.
  Property 'planRequired' does not exist on type 'IntrinsicAttributes & PlanRestrictedFeatureProps'.
Error: src/routes/health.ts(5,26): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
Error: src/routes/health.ts(6,30): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
Error: src/utils/error-tracking.ts(14,21): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
Error: src/utils/error-tracking.ts(27,21): error TS2339: Property 'env' does not exist on type 'ImportMeta'.
Error: src/validation/cross-field-validator.ts(15,50): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/cross-field-validator.ts(36,53): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/cross-field-validator.ts(53,52): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/cross-field-validator.ts(59,44): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/cross-field-validator.ts(71,57): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/cross-field-validator.ts(116,43): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/cross-field-validator.ts(137,44): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/cross-field-validator.ts(176,38): error TS2367: This comparison appears to be unintentional because the types 'string | undefined' and 'boolean' have no overlap.
Error: src/validation/custom-ajv-rules.ts(82,41): error TS2709: Cannot use namespace 'Ajv' as a type.
Error: src/validation/custom-ajv-rules.ts(86,5): error TS2353: Object literal may only specify known properties, and 'strict' does not exist in type 'Options'.
Error: src/validation/custom-ajv-rules.ts(92,14): error TS2345: Argument of type 'import("/home/runner/work/openwebui-config.com/openwebui-config.com/node_modules/ajv/lib/ajv").Ajv' is not assignable to parameter of type 'import("/home/runner/work/openwebui-config.com/openwebui-config.com/node_modules/@rjsf/validator-ajv8/node_modules/ajv/dist/core").default'.
  Type 'Ajv' is missing the following properties from type 'Ajv': opts, logger, scope, schemas, and 17 more.
Error: src/validation/custom-ajv-rules.ts(104,7): error TS2554: Expected 2 arguments, but got 1.
Error: src/validation/custom-ajv-rules.ts(107,23): error TS7006: Parameter 'schemaVal' implicitly has an 'any' type.
Error: src/validation/custom-ajv-rules.ts(122,7): error TS2554: Expected 2 arguments, but got 1.
Error: src/validation/custom-ajv-rules.ts(125,23): error TS7006: Parameter 'schemaVal' implicitly has an 'any' type.
Error: src/validation/custom-ajv-rules.ts(135,7): error TS2554: Expected 2 arguments, but got 1.
Error: src/validation/custom-ajv-rules.ts(138,23): error TS7006: Parameter 'schemaVal' implicitly has an 'any' type.
Error: Process completed with exit code 2.
```

--- a/package.json
+++ b/package.json
@@ -8,16 +8,13 @@
     "build": "vite build",
     "build:analyze": "vite build --mode analyze",
     "preview": "vite preview",
-    
     "deploy": "npm run build && wrangler pages deploy dist",
     "deploy:staging": "npm run build && wrangler pages deploy dist --env staging",
     "deploy:production": "npm run build && wrangler pages deploy dist --env production",
-    
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
     "lint:fix": "eslint . --ext ts,tsx --fix",
     "typecheck": "tsc --noEmit",
     "format": "prettier --write \"src/**/*.{ts,tsx}\"",
-    
     "ci": "npm run typecheck && npm run lint && npm run build",
     "test:build": "npm run build && npm run preview"
   },
@@ -77,6 +74,7 @@
     "@typescript-eslint/eslint-plugin": "^6.0.0",
     "@typescript-eslint/parser": "^6.0.0",
     "@vitejs/plugin-react": "^4.0.3",
+    "ajv": "^8.17.1",
     "autoprefixer": "^10.4.14",
     "eslint": "^8.45.0",
     "eslint-plugin-react-hooks": "^4.6.0",

--- a/src/components/ui/form/feedback/validation-summary.tsx
+++ b/src/components/ui/form/feedback/validation-summary.tsx
@@ -29,10 +29,10 @@ export function ValidationSummary({
     return 'text-green-600'
   }
 
-  const getProgressColor = () => {
-    if (errorCount > 0) return 'bg-red-600'
-    if (warningCount > 0) return 'bg-yellow-600'
-    return 'bg-green-600'
+  const getProgressClassName = () => {
+    if (errorCount > 0) return '[&>*]:bg-red-600'
+    if (warningCount > 0) return '[&>*]:bg-yellow-600'
+    return '[&>*]:bg-green-600'
   }
 
   return (
@@ -52,8 +52,7 @@ export function ValidationSummary({
         
         <Progress 
           value={completionPercentage} 
-          className="h-2"
-          indicatorClassName={getProgressColor()}
+          className={cn("h-2", getProgressClassName())}
         />
         
         <div className="grid grid-cols-3 gap-4 text-center">

--- a/src/form/custom-fields/OverrideableField.tsx
+++ b/src/form/custom-fields/OverrideableField.tsx
@@ -1,5 +1,5 @@
 import React, { useState } from 'react'
-import { FieldProps } from '@rjsf/core'
+import { FieldProps } from '@rjsf/utils'
 import { OverrideableField } from '@/components/ui/form/wrappers/overrideable-field'
 
 export default function OverrideableFieldWrapper(props: FieldProps) {
@@ -9,8 +9,8 @@ export default function OverrideableFieldWrapper(props: FieldProps) {
   
   // If not overrideable, render the default field
   if (!isOverrideable) {
-    const DefaultFieldTemplate = props.registry.templates.FieldTemplate
-    return <DefaultFieldTemplate {...props} />
+    const DefaultField = props.registry.fields.StringField || props.registry.fields.default
+    return <DefaultField {...props} />
   }
   
   const handleOverride = (overridden: boolean) => {
@@ -21,16 +21,17 @@ export default function OverrideableFieldWrapper(props: FieldProps) {
     }
   }
   
-  // Wrap the field with override capability
+  // Use the OverrideableField component with proper props
   return (
     <OverrideableField 
-      defaultValue={defaultValue}
-      onOverride={handleOverride}
-    >
-      {React.createElement(props.registry.templates.FieldTemplate, {
-        ...props,
-        disabled: !isOverridden && defaultValue !== undefined
-      })}
-    </OverrideableField>
+      label={props.schema.title || props.name}
+      description={props.schema.description}
+      defaultValue={String(defaultValue || '')}
+      value={String(props.formData || '')}
+      onChange={(value) => props.onChange(value)}
+      overridden={isOverridden}
+      onOverrideChange={handleOverride}
+      id={props.idSchema.$id}
+    />
   )
 }

--- a/src/form/custom-fields/PlanRestrictedField.tsx
+++ b/src/form/custom-fields/PlanRestrictedField.tsx
@@ -1,5 +1,5 @@
 import React from 'react'
-import { FieldProps } from '@rjsf/core'
+import { FieldProps } from '@rjsf/utils'
 import { PlanRestrictedFeature } from '@/components/ui/form/wrappers/plan-restricted-feature'
 
 export default function PlanRestrictedField(props: FieldProps) {
@@ -7,14 +7,20 @@ export default function PlanRestrictedField(props: FieldProps) {
   
   // If no plan restriction, render the default field
   if (!planRequired) {
-    const DefaultFieldTemplate = props.registry.templates.FieldTemplate
-    return <DefaultFieldTemplate {...props} />
+    const DefaultField = props.registry.fields.StringField || props.registry.fields.default
+    return <DefaultField {...props} />
   }
   
   // Wrap the field with plan restriction
   return (
-    <PlanRestrictedFeature planRequired={planRequired as string}>
-      {React.createElement(props.registry.templates.FieldTemplate, props)}
+    <PlanRestrictedFeature 
+      title={props.schema.title || props.name}
+      description={props.schema.description || ''}
+      requiredPlan={planRequired as string}
+      callToAction="Upgrade to access"
+      onUpgradeClick={() => console.log('Upgrade clicked')}
+    >
+      {React.createElement(props.registry.fields.StringField || props.registry.fields.default, props)}
     </PlanRestrictedFeature>
   )
 }

--- a/src/validation/cross-field-validator.ts
+++ b/src/validation/cross-field-validator.ts
@@ -12,7 +12,7 @@ export class CrossFieldValidator {
     const errors: ValidationResult[] = []
     
     // OpenAI API dependencies
-    if (formData.ENABLE_OPENAI_API === 'true' || formData.ENABLE_OPENAI_API === true) {
+    if (formData.ENABLE_OPENAI_API === 'true' || formData.ENABLE_OPENAI_API === '1') {
       if (!formData.OPENAI_API_KEY && !formData.OPENAI_API_KEYS) {
         errors.push({
           field: 'OPENAI_API_KEY',
@@ -33,7 +33,7 @@ export class CrossFieldValidator {
     }
     
     // Anthropic API dependencies
-    if (formData.ENABLE_ANTHROPIC_API === 'true' || formData.ENABLE_ANTHROPIC_API === true) {
+    if (formData.ENABLE_ANTHROPIC_API === 'true' || formData.ENABLE_ANTHROPIC_API === '1') {
       if (!formData.ANTHROPIC_API_KEY) {
         errors.push({
           field: 'ANTHROPIC_API_KEY',
@@ -50,13 +50,13 @@ export class CrossFieldValidator {
     }
     
     // OAuth dependencies
-    if (formData.ENABLE_OAUTH_SIGNUP === 'true' || formData.ENABLE_OAUTH_SIGNUP === true) {
+    if (formData.ENABLE_OAUTH_SIGNUP === 'true' || formData.ENABLE_OAUTH_SIGNUP === '1') {
       const oauthErrors = this.validateOAuthConfig(formData)
       errors.push(...oauthErrors)
     }
     
     // LDAP dependencies
-    if (formData.ENABLE_LDAP === 'true' || formData.ENABLE_LDAP === true) {
+    if (formData.ENABLE_LDAP === 'true' || formData.ENABLE_LDAP === '1') {
       const ldapErrors = this.validateLDAPConfig(formData)
       errors.push(...ldapErrors)
     }
@@ -68,7 +68,7 @@ export class CrossFieldValidator {
     }
     
     // WebSocket dependencies
-    if (formData.ENABLE_WEBSOCKET_SUPPORT === 'true' || formData.ENABLE_WEBSOCKET_SUPPORT === true) {
+    if (formData.ENABLE_WEBSOCKET_SUPPORT === 'true' || formData.ENABLE_WEBSOCKET_SUPPORT === '1') {
       if (!formData.WEBSOCKET_URL && !formData.WEBSOCKET_MANAGER) {
         errors.push({
           field: 'WEBSOCKET_URL',
@@ -113,7 +113,7 @@ export class CrossFieldValidator {
     }
     
     // Check safe mode
-    if (formData.SAFE_MODE === 'false' || formData.SAFE_MODE === false) {
+    if (formData.SAFE_MODE === 'false' || formData.SAFE_MODE === '0') {
       warnings.push({
         field: 'SAFE_MODE',
         message: 'Safe mode is disabled',
@@ -134,7 +134,7 @@ export class CrossFieldValidator {
     }
     
     // Check authentication
-    if (formData.WEBUI_AUTH === 'false' || formData.WEBUI_AUTH === false) {
+    if (formData.WEBUI_AUTH === 'false' || formData.WEBUI_AUTH === '0') {
       warnings.push({
         field: 'WEBUI_AUTH',
         message: 'Authentication is disabled',
@@ -173,7 +173,7 @@ export class CrossFieldValidator {
     })
     
     // Check debug mode
-    if (formData.DEBUG === 'true' || formData.DEBUG === true) {
+    if (formData.DEBUG === 'true' || formData.DEBUG === '1') {
       issues.push({
         field: 'DEBUG',
         message: 'Debug mode is enabled',

--- a/src/validation/custom-ajv-rules.ts
+++ b/src/validation/custom-ajv-rules.ts
@@ -83,7 +83,6 @@ export function setupCustomValidator(): Ajv {
   const ajv = new Ajv({ 
     allErrors: true, 
     verbose: true,
-    strict: false,
     validateFormats: true,
     addUsedSchema: false
   })
@@ -93,18 +92,14 @@ export function setupCustomValidator(): Ajv {
   
   // Add custom formats
   Object.entries(customFormats).forEach(([name, format]) => {
-    ajv.addFormat(name, {
-      validate: format.validate,
-      // @ts-ignore - AJV types don't include error property but it works
-      error: format.error
-    })
+    ajv.addFormat(name, format.validate)
   })
   
   // Add custom keywords for enhanced validation
   ajv.addKeyword({
     keyword: 'dependsOn',
     schemaType: 'object',
-    compile: function(schemaVal) {
+    compile: function(schemaVal: any) {
       return function validate(data: any, dataPath: any) {
         const parent = dataPath?.parentData
         if (!parent) return true
@@ -122,7 +117,7 @@ export function setupCustomValidator(): Ajv {
   ajv.addKeyword({
     keyword: 'requiresAny',
     schemaType: 'array',
-    compile: function(schemaVal) {
+    compile: function(schemaVal: any) {
       return function validate(data: any, dataPath: any) {
         const parent = dataPath?.parentData
         if (!parent) return true
@@ -135,7 +130,7 @@ export function setupCustomValidator(): Ajv {
   ajv.addKeyword({
     keyword: 'conflictsWith',
     schemaType: ['string', 'array'],
-    compile: function(schemaVal) {
+    compile: function(schemaVal: any) {
       const conflicts = Array.isArray(schemaVal) ? schemaVal : [schemaVal]
       return function validate(data: any, dataPath: any) {
         const parent = dataPath?.parentData

--- a/src/vite-env.d.ts
+++ b/src/vite-env.d.ts
@@ -1,0 +1,11 @@
+/// <reference types="vite/client" />
+
+interface ImportMetaEnv {
+  readonly VITE_APP_VERSION: string
+  readonly VITE_API_BASE_URL: string
+  // Add more env variables as needed
+}
+
+interface ImportMeta {
+  readonly env: ImportMetaEnv
+}

